models/POP/shell_scripts/cesm2_1/CESM2_1_setup_hybrid

```diff
@@ -0,0 +1,766 @@
+   #
+   # README:
+   # Configuring the contents of the history files for POP is best explained in
+   # the cesm2 pop2 namelist documentation.
+   # http://www.cesm.ucar.edu/models/cesm2/settings/current/pop2_nml.html
+   #
+   # and the CESM-specific documentation for the tavg output variables in the pop2
+   # online documentation:
+   # https://ncar.github.io/POP/doc/build/html/users_guide/model-diagnostics-and-output.html#time-averaged-history-files
+
+   echo "init_ts_suboption  = 'data_assim'" >> ${fname}
+
+   @ inst ++
+end
+
+# ==============================================================================
+# Staging restart files now that the run directory exists
+# ==============================================================================
+
+echo ''
+echo 'Staging restart files now that the run directory exists'
+echo ''
+
+set init_time = ${reftimestamp}
+
+cat << EndOfText >! stage_cesm_files
+#!/bin/csh -f
+# This script can be used to help restart an experiment from any previous step.
+# The appropriate files are copied to the RUN directory.
+#
+# Before running this script:
+#  1) be sure CONTINUE_RUN is set correctly in the env_run.xml file in
+#     your CASEROOT directory.
+#     CONTINUE_RUN=FALSE => you are starting over at the initial time.
+#     CONTINUE_RUN=TRUE  => you are starting from a previous step but not
+#                           the very first one.
+#  2) be sure 'restart_time' is set to the day and time that you want to
+#     restart from if not the initial time.
+
+set restart_time = $init_time
+
+# get the settings for this case from the CESM environment
+
+cd \${CASEROOT}
+
+setenv RUNDIR           `./xmlquery RUNDIR           --value`
+setenv CONTINUE_RUN     `./xmlquery CONTINUE_RUN     --value`
+
+cd \${RUNDIR}
+
+echo 'Copying the required CESM files to the run directory to rerun'
+echo 'a previous step.  CONTINUE_RUN from env_run.xml is' \${CONTINUE_RUN}
+if ( \${CONTINUE_RUN} == TRUE ) then
+  echo 'so files for some later step than the initial one will be restaged.'
+  echo "Date to reset files to is: \${restart_time}"
+else
+  echo 'so files for the initial step of this experiment will be restaged.'
+  echo "Date to reset files to is: ${init_time}"
+endif
+echo ''
+
+if ( \${CONTINUE_RUN} == TRUE ) then
+
+   #----------------------------------------------------------------------
+   # This block copies over a set of restart files from any previous step of
+   # the experiment that is NOT the initial step.
+   # After running this script resubmit the job to rerun.
+   #----------------------------------------------------------------------
+
+   echo "Staging restart files for run date/time: " \${restart_time}
+
+   #  The short term archiver is on, so the files we want should be in one
+   #  of the short term archive 'rest' restart directories.  This assumes
+   #  the long term archiver has NOT copied these files to the HPSS yet.
+
+   if (  \${DOUT_S} == TRUE ) then
```

Comment:
DOUT_S should be set from a .xmlquery call ... similar to RUNDIR and CONTINUE_RUN.  You can turn short-term archiving on or off during the course of a run, and you might need to run stage_cesm_files more than just at the beginning.